### PR TITLE
#TP-22 ft:(resetPasword) forgot password and reset password

### PR DESCRIPTION
**Pull request by Delyc**
### What does this PR do?
Allowing a user to reset password

### Description of Task to be completed

- [x] The driver or operator should be able to receive a reset password link in their email.
- [x] The driver or operator should be able to reset the password( i.e. provide a new password).
- [x] Appropriate error messages are displayed on unsuccessful attempts
- [x] The driver or operator should be able to login with the new password.

### How should this be manually tested?

```
git clone https://github.com/atlp-rwanda/phatom-be-techode
cd phantom-be-techode
git checkout ft-create-reset-password-tp-22
npm install
npm run migrate
npm run dev

//after npm run dev check database connection, if it is successful go ahead and test in postman otherwise start up database

```

### Any background context you want to provide?
When sending email I have used a package called **Nodemailer**
N:B for this to run smoothly on your computer, you need to have:
```
EMAIL_USER=<sender's email>
EMAIL_PASS=<password of sender's email>
SERVICE=<service being used>
``` 
 
### What are the relevant Trello stories?
[TP-22](https://trello.com/c/4mPP2nHp/22-reset-password)

[Finishes #TP-22]

### CODE REVIEW

**Review by wakaflorien — APPROVED** [review]

**Review by chrisostome01 — APPROVED** [review]
### Well done guys

> Now we have perfect coverage ;)

File: controllers/resetController.js [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+import { users, resetTokens } from '../models';
+import db from '../models/index.js';
+import sendMail from './resetUtil.js';
+import { success, fail, sendError } from '../function/respond.js';
+
+export const forgotPassword = async (req, res) => {
+	try {
+		const { email } = req.body;
+		if (!req.body.email) {
+			return fail(res, 401, null, 'missingEmail', req);
```
Comment (Jkadhuwa) [inline controllers/resetController.js:11]: Please use the correct status codes. 
https://developer.mozilla.org/en-US/docs/Web/HTTP/Status

File: test/resetPassword.test.js [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+	});
+	it('should create a User', async () => {
+		const user = {
+			fullname: 'cyifuzo jean chrysostome',
+			username: 'chance',
+			password: 'test123',
+			email: 'delcy@gmail.com',
+		};
+		const response = await chai.request(app).post(`/api/v1/users`).send(user);
+		expect(response).to.have.status(201);
```
Comment (Jkadhuwa) [inline test/resetPassword.test.js:28]: You can add a few more assertions, to see if you are getting the right response.

**Review by Jkadhuwa — DISMISSED** [review]
Nice work @Delyc and team, left some feedback for your action.

**Review by chanceDira — APPROVED** [review]

**Pull request merged**